### PR TITLE
Bug 1156185 - [Stingray][TvDeck] Add onscanningstatechange...

### PR DESCRIPTION
**Pull request by sean2449**
...d event by using addEventListener

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=c0191e83fb06356459cae57de53856f250110237

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9a551e816aad05d8aec313945e006874ec630417

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5bb58bc6fa7402c6e1b83d47fc173ef0c4a8adc6

File: tv_apps/tv-deck/js/tv_deck.js [hunk 1]
```diff
@@ -297,21 +274,21 @@
...
+    source.addEventListener(
+              'scanningstatechanged', function onscanningstatechanged(event) {
+      var state = event.state;
+      switch (state) {
+        case 'completed':
+        /* falls through */
+        case 'stopped':
+          this._onScanningCompleted();
+          source.removeEventListener(
+                                'scanningstatechanged', onscanningstatechanged);
```
Comment (huchengtw-moz) [inline tv_apps/tv-deck/js/tv_deck.js]: The function you reference here is not the one you added to event listener. Once bind() called, it returns another function. So, you cannot remove the event listener using this way.

File: tv_apps/tv-deck/js/tv_deck.js [hunk 2]
```diff
@@ -108,29 +108,6 @@
         this.manifestURL = evt.target.result.manifestURL;
       }
     }.bind(this);
-
-    // Set the 'ontunerchanged' event handler.
-    navigator.tv.ontunerchanged = function(event) {
```
Comment (huchengtw-moz) [inline tv_apps/tv-deck/js/tv_deck.js]: Where is turnerchanged event handler? You had remove it but not add it back.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=73473cd74992ef98bfbf3743d345026770c22f4b

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5dbad95205743f0bd28b4bee584843be9412be5e

**Pull request merged**